common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -775,12 +775,18 @@ public void execute(Runnable task) {
         }
 
         boolean inEventLoop = inEventLoop();
-        addTask(task);
         if (!inEventLoop) {
+            if (isShutdown()) {
```

Comment:
Yes this makes things less racy. Why not use a CAS loop here to make even less racy (especially as we have a  volatile field updater available)?

Reply:
I see the `shutdown` method is `@Deprecated`. Is that to remove it from the public API but it will still be called by something? A bit of javadoc about the deprecation would be great.

Reply:
oh duh! you are right...